### PR TITLE
external-dns: allow custom sa

### PR DESCRIPTION
**Pull request by mcharriere**
### What does this PR do?

Create external-dns category and add GCP custom SA docs

### What does it look like?

image

image

### Any background context you can provide?

Towards https://github.com/giantswarm/roadmap/issues/1731

### Have you maintained the front matter?

(Please bump the last_review_date in case this qualifies as a review for an entire page. Provide user_questions which should be answered in the page. Provide a meaningful description.)

### CODE REVIEW

File: src/content/advanced/external-dns/gcp-custom-sa.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+last_review_date: 2022-12-21
+user_questions:
+  - How can I override GCP service account used by External DNS?
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-cabbage
+---
+
+## Overview
+
+By default, our External DNS App uses the `gcpProject` value to compose the right Service account annotation to correctly authenticate against GCP.
```
Comment (Gacko) [inline src/content/advanced/external-dns/gcp-custom-sa.md:18]:
```suggestion
By default, our External DNS App uses the `gcpProject` value to compose the right service account annotation to correctly authenticate against GCP.
```

I'd propose either `ServiceAccount` or just "service account".

File: src/content/advanced/external-dns/gcp-custom-sa.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+linkTitle: Using a dedicated GCP service account
+title: External DNS using a dedicated GCP service account
+description: This document describes how to configure External DNS to use a dedicated GCP Service Account.
```
Comment (Gacko) [inline src/content/advanced/external-dns/gcp-custom-sa.md:4]:
```suggestion
description: This document describes how to configure External DNS to use a dedicated GCP service account.
```

Should be same as in user questions, right?

File: src/content/advanced/external-dns/gcp-custom-sa.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  - https://github.com/orgs/giantswarm/teams/team-cabbage
+---
+
+## Overview
+
+By default, our External DNS App uses the `gcpProject` value to compose the right Service account annotation to correctly authenticate against GCP.
+
+## Configuration
+
+To override the default configuration, use the following values with the external-dns-app:
```
Comment (Gacko) [inline src/content/advanced/external-dns/gcp-custom-sa.md:22]:
```suggestion
To override the default configuration, use the following values with the `external-dns-app`:
```

... or just "External DNS App" to have it consistent with other mentions, your choice.

**Review by Gacko — CHANGES_REQUESTED** [review]

File: src/content/advanced/external-dns/gcp-custom-sa.md (hunk 1 above)
Comment (mleisa) [reply src/content/advanced/external-dns/gcp-custom-sa.md:18]: In Google Cloud documentation, they refer to it as `service account`. Unless we use it differently, I'd vote we call it the same

Comment (mcharriere) [reply src/content/advanced/external-dns/gcp-custom-sa.md:18]: this refers to the Kubernetes service account, so I'd go for `ServiceAccount`

Comment (mcharriere) [reply src/content/advanced/external-dns/gcp-custom-sa.md:18]:
```suggestion
By default, our External DNS App uses the `gcpProject` value to compose the right annotation for the `ServiceAccount` to correctly authenticate against GCP.
```

**Review by Gacko — APPROVED** [review]
Looks great, thank you!

**Pull request merged**